barretenberg/cpp/src/barretenberg/stdlib/plonk_recursion/aggregation_state/aggregation_state.hpp

```diff
@@ -92,18 +111,102 @@ template <typename Curve> void write(std::vector<uint8_t>& buf, aggregation_stat
 
     write(buf, as.P0);
     write(buf, as.P1);
-    write(buf, as.public_inputs);
-    write(buf, as.proof_witness_indices);
+    // write(buf, as.public_inputs);
+    // write(buf, as.proof_witness_indices);
     write(buf, as.has_data);
 };
 
 template <typename NCT> std::ostream& operator<<(std::ostream& os, aggregation_state<NCT> const& as)
 {
     return os << "P0: " << as.P0 << "\n"
-              << "P1: " << as.P1 << "\n"
-              << "public_inputs: " << as.public_inputs << "\n"
-              << "proof_witness_indices: " << as.proof_witness_indices << "\n"
+              << "P1: " << as.P1
+              << "\n"
+              //   << "public_inputs: " << as.public_inputs << "\n"
+              //   << "proof_witness_indices: " << as.proof_witness_indices << "\n"
               << "has_data: " << as.has_data << "\n";
 }
+template <typename Builder, typename Curve>
+aggregation_state<Curve> convert_witness_indices_to_agg_obj(Builder& builder,
```

Comment:
creates an aggregation object given a 16 valid witness indices to the bigfield limbs of the 4 bigfields of the aggregation object.